File: day1/day1_2.py
f = open("day1.txt", "r")
data = f.read()
lines = data.split('\n')

used_lines = filter(lambda line: len(line) > 0 and not line.isspace(), lines)
expenses = map(int, used_lines)

for i in range(0, len(expenses)):
    for j in range(i+1, len(expenses)):
        for k in range(j+1, len(expenses)):
            if expenses[i] + expenses[j] + expenses[k] == 2020:
                print(expenses[i] * expenses[j] * expenses[k])
                exit()

File: day1/day1_2.rs
use std::io::Read;
use std::io;
use std::fs::File;


fn read_file(file_name: &str) -> Vec<String> {
    let mut file = File::open(file_name).unwrap();

    let mut string = String::new();
    file.read_to_string(&mut string).unwrap();

    let lines = string.lines();

    return lines.map(|s| s.to_string()).collect::<Vec<String>>();
}

fn main() {
    let lines = read_file("day1.txt");

    let expenses = lines.iter()
                        .filter(|s| !s.is_empty())
                        .map(|s| s.parse::<i64>().unwrap())
                        .collect::<Vec<i64>>();

    for expense1 in expenses.iter() {
        for expense2 in expenses.iter() {
            for expense3 in expenses.iter() {
                if expense1 + expense2 + expense3 == 2020 {
                    println!("{}", expense1 * expense2 * expense3 );
                    return;
                }
            }
        }
    }
    println!("Something went wrong...");
}
